lib/ansible/module_utils/facts.py

```diff
@@ -1208,58 +1213,122 @@ def get_dmi_facts(self):
+        findmnt_path = self.module.get_bin_path("findmnt")
+        if not findmnt_path:
+            return
+
+        rc, out, err = self._run_findmnt(findmnt_path)
+        if rc != 0:
+            return bind_mounts
+
+        # find bind mounts, in case /etc/mtab is a symlink to /proc/mounts
+        lines = out.splitlines()
+        for line in lines:
+            fields = line.split()
+            # fields[0] is the TARGET, fields[1] is the SOURCE
+            if len(fields) < 2:
+                continue
+
+            # bind mounts will have a [/directory_name] in the SOURCE column
+            if re.match(".*\]",fields[1]):
+                bind_mounts.append(fields[0])
 
+        return bind_mounts
+
+    def _mtab_entries(self):
         mtab = get_file_content('/etc/mtab', '')
-        for line in mtab.split('\n'):
+        lines = mtab.splitlines()
+        mtab_entries = []
+        for line in lines:
             fields = line.rstrip('\n').split()
             if len(fields) < 4:
                 continue
-            if fields[0].startswith('/') or ':/' in fields[0]:
-                if(fields[2] != 'none'):
-                    size_total, size_available = self._get_mount_size_facts(fields[1])
-                    if fields[0] in uuids:
-                        uuid = uuids[fields[0]]
-                    else:
-                        uuid = 'NA'
-                        lsblkPath = self.module.get_bin_path("lsblk")
-                        if lsblkPath:
-                            rc, out, err = self.module.run_command("%s -ln --output UUID %s" % (lsblkPath, fields[0]), use_unsafe_shell=True)
-
-                            if rc == 0:
-                                uuid = out.strip()
-                                uuids[fields[0]] = uuid
-
-                    if fields[1] in bind_mounts:
-                        # only add if not already there, we might have a plain /etc/mtab
-                        if not re.match(".*bind.*", fields[3]):
-                            fields[3] += ",bind"
-
-                    self.facts['mounts'].append(
-                        {'mount': fields[1],
-                         'device':fields[0],
-                         'fstype': fields[2],
-                         'options': fields[3],
-                         # statvfs data
-                         'size_total': size_total,
-                         'size_available': size_available,
-                         'uuid': uuid,
-                         })
+            mtab_entries.append(fields)
+        return mtab_entries
+
+    @timeout(10)
+    def get_mount_facts(self):
+        self.facts['mounts'] = []
+
+        bind_mounts = self._find_bind_mounts()
+        uuids = self._lsblk_uuid()
+        mtab_entries = self._mtab_entries()
+
+        mounts = []
+        for fields in mtab_entries:
+            device, mount, fstype, options = fields[0], fields[1], fields[2], fields[3]
```

Comment:
This is to guard against the size of fields changing and causing tuple unpacking errors.
 _mtab_entries() will potentially return entries with > 4 items.
Though the code to parse the entries  is already a bit split across _mtab_entries and get_mount_facts so that could be enforced there.